src/py/mat3ra/made/tools/analyze/interface.py

```diff
@@ -22,51 +57,164 @@ def substrate_material(self):
+            film_strain2d = film_sl @ np.linalg.inv(film2d @ film_trans)
+
+            sub_conf = SlabStrainedSupercellConfiguration(
+                stack_components=self.substrate_slab_configuration.stack_components,
+                direction=self.substrate_slab_configuration.direction,
+                xy_supercell_matrix=SupercellMatrix2DSchema(root=sub_trans.astype(int).tolist()),
+                strain_matrix=self.to_3d_strain(sub_strain2d),
+            )
+            film_conf = SlabStrainedSupercellConfiguration(
+                stack_components=self.film_slab_configuration.stack_components,
+                direction=self.film_slab_configuration.direction,
+                xy_supercell_matrix=SupercellMatrix2DSchema(root=film_trans.astype(int).tolist()),
+                strain_matrix=self.to_3d_strain(film_strain2d),
+            )
+            configs.append((sub_conf, film_conf))
+
+        return configs
+
+    def get_strained_slab_configurations_with_metadata(
+        self,
+    ) -> List[ZSLConfigurationWithStrain]:
+        sub2d = self.to_2d(self.substrate_material.lattice)
+        film2d = self.to_2d(self.film_material.lattice)
+        sub_area = vec_area(sub2d[0], sub2d[1])
+        film_area = vec_area(film2d[0], film2d[1])
+        results: List[ZSLConfigurationWithStrain] = []
+
+        sets = self._zsl.generate_sl_transformation_sets(film_area, sub_area)
+        matches = self._zsl.get_equiv_transformations(sets, film2d, sub2d)
+
+        for film_sl, sub_sl, film_trans, sub_trans in matches:
+            sub_strain2d = sub_sl @ np.linalg.inv(sub2d @ sub_trans)
+            film_strain2d = film_sl @ np.linalg.inv(film2d @ film_trans)
+
+            sub_conf = SlabStrainedSupercellConfiguration(
+                stack_components=self.substrate_slab_configuration.stack_components,
+                direction=self.substrate_slab_configuration.direction,
+                xy_supercell_matrix=SupercellMatrix2DSchema(root=sub_trans.astype(int).tolist()),
+                strain_matrix=self.to_3d_strain(sub_strain2d),
+            )
+            film_conf = SlabStrainedSupercellConfiguration(
+                stack_components=self.film_slab_configuration.stack_components,
+                direction=self.film_slab_configuration.direction,
+                xy_supercell_matrix=SupercellMatrix2DSchema(root=film_trans.astype(int).tolist()),
+                strain_matrix=self.to_3d_strain(film_strain2d),
+            )
+            info = compute_strains_from_match(sub_sl, film_sl, sub2d, film2d)
+            results.append(
+                ZSLConfigurationWithStrain(
+                    substrate_config=sub_conf,
+                    film_config=film_conf,
+                    strain_info=info,
+                )
+            )
+
+        # sort by average total strain
+        results.sort(key=lambda x: (x.strain_info["substrate_strain"] + x.strain_info["film_strain"]) / 2)
+        return results
```

Comment:
Let's add SlabPairHolder and build it just like we built `termination_holders` in CrystalLatticePlanesMaterialAnalyzer

Reply:
Define a TerminationHolder analog, `termination_holders` property analog